File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mn=1e5,ret=0;
        for(auto it : prices)
        {
            ret=max(ret,it-mn);
            mn=min(mn,it);
        }
        return ret;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
   public int maxProfit(int[] prices) {
        int mx = 0, num = 0;
        for(int i = prices.length - 1;i >= 0; i--){
            mx = Math.max(mx, num - prices[i]);
            if(num< prices[i])
                num = prices[i];
        }
        return mx;
    }
}